android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt

```diff
@@ -29,7 +59,58 @@ class LoginViewModel @Inject constructor(private val repo: ParkEasyRepository) :
         _password.value = input
     }
 
+    fun onConfirmPasswordChange(input: String) {
+        _confirmPassword.value = input
+        _matchingPasswords.value = _password.value == _confirmPassword.value
+    }
+
     fun onLoginPress() {
-        runBlocking { login(Credentials(_email.value, _password.value)) }
+        runBlocking {
+            launch {
+                if (userRepo.login(LoginCredentials(_email.value, _password.value))) {
+                    _loggedIn.value = true
+                    _message.value = Event("Logged in successfully")
+                } else {
+                    _message.value = Event("Error logging in")
+                }
+            }
+        }
+    }
+
+    fun onRegisterPress() {
+        if (_password.value == _confirmPassword.value) {
+            runBlocking {
+                launch {
+                    if (
+                        userRepo.register(
+                            RegistrationCredentials(_name.value, _email.value, _password.value)
+                        )
+                    ) {
+                        _loggedIn.value = true
+                        _message.value = Event("Registered successfully")
+                    } else {
+                        _message.value = Event("Error registering")
```

Comment:
For registration, getting the error code from the response is better. If you don't want to throw or catch exception, I suggest creating a Response object outside of the call and pass it in (kinda like what we do with pointer in C)